ibis/backends/tests/test_client.py

```diff
@@ -278,6 +277,7 @@ def test_rename_table(con, temp_table, temp_table_orig, new_schema):
 @mark.notyet(
     ["trino"], reason="trino doesn't support NOT NULL in its in-memory catalog"
 )
+@mark.broken(["snowflake"], reason="snowflake shows not nullable column as nullable")
```

Comment:
Is this broken upstream or is there a bug in the implementation of type parsing for snowflake?

Reply:
This is due to a weird behaviour of snowflake

```
>create temporary table _ibis_test_nullability (a number not null);
+----------------------------------------------------+
| status                                             |
|----------------------------------------------------|
| Table _IBIS_TEST_NULLABILITY successfully created. |
+----------------------------------------------------+
1 Row(s) produced. Time Elapsed: 0.470s

>desc _ibis_test_nullability;
+------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------+
| name | type         | kind   | null? | default | primary key | unique key | check | expression | comment | policy name |
|------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------|
| A    | NUMBER(38,0) | COLUMN | N     | NULL    | N           | N          | NULL  | NULL       | NULL    | NULL        |
+------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------+
1 Row(s) produced. Time Elapsed: 0.212s

>select * from _ibis_test_nullability;
+---+
| A |
|---|
+---+
0 Row(s) produced. Time Elapsed: 0.224s

>DESC RESULT last_query_id();
+------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------+
| name | type         | kind   | null? | default | primary key | unique key | check | expression | comment | policy name |
|------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------|
| A    | NUMBER(38,0) | COLUMN | Y     | NULL    | N           | N          | NULL  | NULL       | NULL    | NULL        |
+------+--------------+--------+-------+---------+-------------+------------+-------+------------+---------+-------------+
1 Row(s) produced. Time Elapsed: 0.222s

```

Reply:
See the `null?` values.